tools/ci_build/build.py

```diff
@@ -569,6 +578,20 @@ def setup_test_data(build_dir, configs):
                                 src_model_dir], shell=True)
 
 
+def use_dev_mode(args):
+    if args.use_acl or args.use_armnn:
+        return 'OFF'
+    if is_macOS():
+        if args.ios:
+            return 'OFF'
+        if args.enable_pybind and is_using_xcode_12():
+            # pybind11 warning if use AppleClang 12 to compile
+            # https://github.com/pybind/pybind11/pull/2522
+            # TODO, remove this after switch to pybind11 2.6.0+
+            return 'OFF'
+    return 'ON'
```

Comment:
Disabling dev mode feels like a fairly severe hack vs. either fixing warnings where possible or disabling specific warnings on specific targets. Whilst this is consistent with the existing approach, it would be good to have a more refined way to handle warnings on iOS/macOS builds than this. Not sure why ACL or ARMNN should need this either. Could you maybe add a task to the ORT Core backlog for someone to investigate improving this?

Reply:
Create 3 PBI
Mobile
- Product Backlog Item 920207: Enable dev mode for ORT iOS
Core
- Product Backlog Item 920213: Investigate enabling dev mode for ACL/ARMNN EP
- Product Backlog Item 920214: Enable dev mode for building with xcode 12 on macOS after update pybind11 to 2.6.0+